common/common.go

```diff
@@ -102,4 +102,8 @@ var ArgoCDManagerPolicyRules = []rbacv1.PolicyRule{
 		Resources: []string{"*"},
 		Verbs:     []string{"*"},
 	},
+	{
+		NonResourceURLs: []string{"/metrics"},
+		Verbs:           []string{"get"},
```

Comment:
Given that argocd is supposed to be able to create/manage everything in a cluster, I think we should just make these privileges equivalent to cluster-admin and give `*`, `*` to nonResourceURLs / verbs. Otherwise, we'll just end up running into a similar problem when deploying a different app with a different URL.

Reply:
Good point, I will update the pull request. 
Also thanks for the project, it is a great approach to gitops.

Reply:
Hi @jessesuen I think it's not fair to assume that we would have cluster-admin privilege when trying to use argocd for deployment. Cluster could be shared with multiple tenants and admin privileges could be scoped to a namespace or group. Could we make `argocd-manager-role` customizable?